File: backend/app/models/note.py
import datetime

from sqlalchemy import (
    Column, Integer, String, Text, DateTime,
    ForeignKey, func,
)

from app.db.base import Base


class Note(Base):
    __tablename__ = "notes"

    id = Column(Integer, primary_key=True, autoincrement=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False, index=True)
    document_id = Column(Integer, ForeignKey("documents.id"), nullable=True)
    source_annotation_id = Column(Integer, nullable=True)
    title = Column(String(256), default="")
    content = Column(Text, default="")
    content_type = Column(String(16), default="markdown")  # markdown | rich_text
    tags = Column(String(256), default="")
    deleted_at = Column(DateTime, nullable=True)
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())

File: backend-go/internal/models/note.go
package models

import "time"

type Note struct {
	ID                 uint      `gorm:"primaryKey;autoIncrement" json:"id"`
	UserID             uint      `gorm:"index;not null" json:"user_id"`
	DocumentID         *uint     `gorm:"index" json:"document_id,omitempty"`
	SourceAnnotationID *uint     `json:"source_annotation_id,omitempty"`
	Title              string    `gorm:"size:256;default:''" json:"title"`
	Content            string    `gorm:"type:text" json:"content"`
	ContentType        string    `gorm:"size:16;default:'markdown'" json:"content_type"`
	Tags               string    `gorm:"size:256;default:''" json:"-"`
	DeletedAt          *time.Time `gorm:"index" json:"-"`
	CreatedAt          time.Time  `json:"created_at"`
	UpdatedAt          time.Time  `json:"updated_at"`

	// Computed fields
	DocumentTitle *string `gorm:"-" json:"document_title,omitempty"`
	TagList       []string `gorm:"-" json:"tags"`
}
